clients/src/main/resources/common/message/ProduceResponse.json

```diff
@@ -32,7 +32,9 @@
   // records that cause the whole batch to be dropped.  See KIP-467 for details.
   //
   // Version 9 enables flexible versions.
-  "validVersions": "0-9",
+  //
+  // Version 10 adds 'CurrentLeader' and 'NodeEndpoints' as tagged fields
```

Comment:
Why do we need to bump the version if we are just adding tagged fields?

Reply:
I'm not sure if this is absolutely necessary, I was going based off the KIP, but I do think there could be an issue with leaving the version the same. If a client is still using the old protocol definition and the server returns a message based on the new definition but with the same version number, I think the client would deserialize it incorrectly?

Reply:
New tagged fields would be unknown to older clients, so they would ignore them. It would not affect their ability to deserialize.

Reply:
ok, I'm fine with keeping the version the same, but we should update the KIP. @msn-tldr do you see any issues with this?

Reply:
I didn't realize this was brought up in previous discussions, it looks like we decided to bump up the version # to make it clearer which clients have implemented the feature, in an email from @dajac 

> Personally, I would rather prefer to bump both versions and to add the
tagged fields. This would allow us to better reason about what the client
is supposed to do when we see the version on the server side. Otherwise, we
will never know if the client uses this or not.

@hachikuji does this sound good?

Reply:
I was also wondering this and found the same discussion on the mailing thread. Ismael also +1'd this comment.

I think generally we should have a strategy with respect to tagged fields. Should we always use the lowest tagged version? Should we only do that if older clients can use it/we backport changes for it?

If we plan to bump on tagged fields is the only benefit that we save some bites over the wire in the case we don't always include the value in the response? (I'm also a bit unsure about the "saving bytes" here since we would just have an empty node endpoint array)

Reply:
I actually pushed for this. I wanted to know when a client supports this new mechanism or not. Without a version bump, it is hard to tell.

Reply:
The tricky thing is that bumping the protocol version does not imply client support (the client can ignore tagged fields). But it does help for the standard client since we know the feature will be supported. That probably does make request analysis easier. So I guess I'm appeased.

Reply:
Are we in agreement then that we should always bump the tagged field then? Unless we explicitly plan to backport the changes (ie what Jeff did for the consumer group changes?)

I had some folks ask me what the correct protocol is, and it would be nice to give consistent answers 😅

Reply:
@jolshan +1

Reply:
@hachikuji Yeah, your point is totally valid. I was pushing for this with the java client (and potentially librdkafka) in mind. I think that it will make request analysis easier as you said.

Reply:
@jolshan I think that it is hard to really define a policy for this. It mainly depends on whether there is a justification to require an epoch bump or not. In this case, I believe that there is one but this may not always be true.

Reply:
I'm also wondering if there is a flaw with this approach. This would mean we need to bump MV after all for inter broker requests right? https://github.com/apache/kafka/pull/14444#discussion_r1357181404

Reply:
I bumped the MV for fetch requests in 3.7, I think it's fine since it isn't released yet, lmk if you see any issues.